boards/arm/stm32h7/nucleo-h743zi/src/stm32_boot.c

```diff
@@ -86,7 +86,7 @@ void stm32_boardinitialize(void)
 #ifdef CONFIG_BOARD_LATE_INITIALIZE
 void board_late_initialize(void)
 {
-#if defined(CONFIG_NSH_LIBRARY) && !defined(CONFIG_BOARDCTL)
+#if defined(CONFIG_NSH_LIBRARY) && !defined(CONFIG_LIB_BOARDCTL)
```

Comment:
seems like `LIB_BOARDCTL` is leftover from renaming. Probably we need to clean-up all the places and do not introduce new dependencies

Reply:
Hi @pkarashchenko, what do you propose in this case? 

I see that there exists other boards does not check neither `CONFIG_NSH_LIBRARY` nor `CONFIG_BOARDCTL` for intializing the board and other that checks both. 

* nuttx/boards/arm/samv7/samv71-xult/src/sam_boot.c
* nuttx/boards/arm/stm32/nucleo-f303ze/src/stm32_boot.c

In my case I have defined `CONFIG_NSH_LIBRARY` and also `CONFIG_BOARDCTL` so the `stm32_bringup()` function is never reachable.

Why `CONFIG_BOARDCTL` must not be defined in order to perform a board_late_initialization?

I am confused about the difference between `CONFIG_BOARDCTL` and `CONFIG_LIB_BOARDCTL`

Reply:
Regarding the implementation of the board_late_initialize I have find that there exist several implementations of the same as resumed in the next diagram:

![Untitled Diagram drawio](https://user-images.githubusercontent.com/25639961/197418606-c64d0124-e10b-42b7-a549-ac3cc14c26f9.png)

Where the different color paths are taken depending on the build caps used.

Maybe an effort should be done here in homogenizing this as it is really difficult to understand the proper way to initialize the board.

Reply:
I will try to take a look here this week

Reply:
What configuration are you using? Maybe you can post a defconfig file, so it will be more easy to track down the root cause

Reply:
Sorry an error was made by me and the PR is gonna be closed. 

The `stm32_bringup()` function is better to be call from app so this PR approach is erroneous.